framework/src/meshmodifiers/BreakMeshByBlockBase.C

```diff
@@ -0,0 +1,139 @@
+                        false,
+                        "If true, it create a "
+                        "different interface for each block pair");
+  params.addClassDescription("This is the base class used to split a monolithic"
+                             "mesh by blocks pairs");
+  return params;
+}
+
+BreakMeshByBlockBase::BreakMeshByBlockBase(const InputParameters & parameters)
+  : MeshModifier(parameters),
+    _interface_name(getParam<std::string>("interface_name")),
+    _split_interface(getParam<bool>("split_interface"))
+{
+}
+
+void
+BreakMeshByBlockBase::modify()
+{
+  mooseError("BreakMeshByBlockBase should never be called directly!"
+             "Always use one of its the derived classes");
+}
+
+void
+BreakMeshByBlockBase::checkInputParameter()
+{
+  // check input consistency
+  if (getParam<bool>("split_interface") &&
+      (_pars.isParamSetByUser("interface_id") || _pars.isParamSetByUser("interface_name")))
+  {
+    mooseError("if split_interface == true, interface_name and/or interface_id"
+               " cannot be specified by the user");
+  }
+}
+
+BoundaryID
+BreakMeshByBlockBase::findFreeBoundaryId()
+{
+  const std::set<BoundaryID> & currentBoundaryIds = _mesh_ptr->getBoundaryIDs();
+  bool freeBoundaryNotFound = true;
+  BoundaryID freeId;
+  for (freeId = 0; freeId < std::numeric_limits<BoundaryID>::max(); freeId++)
+  {
+    if (currentBoundaryIds.count(freeId) == 0)
+    {
+      // bid is not in the set, boundaryID is free
+      freeBoundaryNotFound = false;
+      break;
+    }
+  }
+
+  if (freeBoundaryNotFound)
+    mooseError("Too many boundaries. Maximum limit exceeded!");
+
+  return freeId;
+}
+
+std::string
+BreakMeshByBlockBase::generateBoundaryName(const subdomain_id_type & masterBlockID,
+                                           const subdomain_id_type & slaveBlockID)
+{
+  std::string master_block_name = _mesh_ptr->getSubdomainName(masterBlockID);
+  std::string slave_block_name = _mesh_ptr->getSubdomainName(slaveBlockID);
+  if (master_block_name.empty())
+    master_block_name = "Block" + std::to_string(masterBlockID);
+  if (slave_block_name.empty())
+    slave_block_name = "Block" + std::to_string(slaveBlockID);
+
+  return master_block_name + "_" + slave_block_name;
+}
+
+void
+BreakMeshByBlockBase::mapBoundaryIdAndBoundaryName(BoundaryID & boundaryID,
+                                                   std::string & boundaryName)
+{
+  _bName_bID_set.insert(std::pair<std::string, int>(boundaryName, boundaryID));
+}
+
+void
+BreakMeshByBlockBase::findBoundaryNameAndInd(const subdomain_id_type & masterBlockID,
+                                             const subdomain_id_type & slaveBlockID,
+                                             std::string & boundaryName,
+                                             BoundaryID & boundaryID,
+                                             BoundaryInfo & boundary_info)
+{
+  // TODO this method should be multiprocessor and multithread safe
```

Comment:
This will work with ReplicatedMesh without any special code but needs work for DistributedMesh (you don't have to necessarily make this work with DistributedMesh, that can often be a chore).